src/components/App/MainMenu/link-types/JoinedEventLink.tsx

```diff
@@ -0,0 +1,37 @@
+import { useRouter } from "next/router";
+import { useCallback } from "react";
+import { useCurrentMappingEvent } from "../../../../lib/context/useCurrentMappingEvent";
+import Spinner from "../../../ActivityIndicator/Spinner";
+import { useAppStateAwareHref } from "../../useAppStateAwareHref";
+import type { IAutoLinkProps } from "./AutoLink";
+import MenuItemOrButton from "./MenuItemOrButton";
+import { Text } from "@radix-ui/themes";
+
+/**
+ * Links to the current mapping event, or the events overview if no event is joined.
+ */
+function JoinedEventLink(props: IAutoLinkProps) {
+  const { data: joinedMappingEvent, isValidating } = useCurrentMappingEvent();
+
+  const href = joinedMappingEvent
+    ? `/events/${joinedMappingEvent._id}`
+    : "/events";
+
+  const label = joinedMappingEvent ? joinedMappingEvent.name : props.label;
+  const children = isValidating ? <Spinner /> : <Text>{label}</Text>;
+  const router = useRouter();
+  // XXX: This should be an <AppStateAwareLink>, but this would break keyboard navigation:
+  // A bug in Radix UI (?) appears to not update the internal `ref` correctly.
```

Comment:
**Non-Blocker:** Maybe the AppStateLink needs to use forwardRef in order to be able to be used with ref? Idk, my knowledge of modern react is still a bit limited, but that came to mind. Nothing that we need to solve in this PR.

Reply:
That was it! I was 100% sure I had tried this, but missed it.